service/src/main/java/uk/gov/mca/beacons/api/registration/application/RegistrationReadOnlyService.java

```diff
@@ -79,7 +81,66 @@ public Registration getByBeaconId(BeaconId beaconId) {
       .build();
   }
 
-  public List<Note> getNonSyetemNotesByBeaconId(BeaconId beaconId) {
+  public Registration getRegistrationFromBeaconBackupItem(
+    BeaconBackupItem beaconBackupItem
+  ) {
+    Registration registration = Registration.builder().build();
+    registration =
+      hydrateBeaconFromBeaconBackupItem(registration, beaconBackupItem);
+    BeaconId beaconId = registration.getBeacon().getId();
+
+    BeaconOwner beaconOwner = beaconOwnerRepository
+      .findBeaconOwnerByBeaconId(beaconId)
+      .orElse(null);
+    List<BeaconUse> beaconUses = beaconUseRepository.findBeaconUsesByBeaconId(
+      beaconId
+    );
+    List<EmergencyContact> emergencyContacts = emergencyContactRepository.findEmergencyContactsByBeaconId(
+      beaconId
+    );
+
+    registration.setBeaconOwner(beaconOwner);
+    registration.setBeaconUses(beaconUses);
+    registration.setEmergencyContacts(emergencyContacts);
+
+    return registration;
+  }
+
+  private Registration hydrateBeaconFromBeaconBackupItem(
```

Comment:
Id probably just have this as Beacon buildBeaconFromBackupItem

then registration.setBeacon(buildBeaconFromBackupItem(backupItem)) in the above method